galaxy_ng/app/tasks/promotion.py

```diff
@@ -25,3 +26,35 @@ def remove_content_from_repository(collection_version_pk, repository_pk):
     qs = CollectionVersion.objects.filter(pk=collection_version_pk)
     with repository.new_version() as new_version:
         new_version.remove_content(qs)
+
+
+def dispatch_tasks_to_org_repos(collection_version_pk, action):
+    """Starts tasks to add/remove CollectionVersion to/from org repos.
+    Method to start tasks, not a task itself."""
+
+    # get all org distros
+    # TODO: replace with correct convention, regex in constants?
+    org_distros = AnsibleDistribution.objects.filter(name__startswith='org-')
+
+    # filter out distros which point to golden repo
+    golden = AnsibleRepository.objects.get(name='automation-hub')  # use constant or param
+    org_repos = set([d.repository for d in org_distros])
+    try:
+        org_repos.remove(golden)
+    except KeyError:
+        pass
+
+    if action == 'remove':
+        for repo in org_repos:
+            locks = [repo]
+            task_args = (collection_version_pk, repo.pk)
+            enqueue_with_reservation(remove_content_from_repository, locks, args=task_args)
+    elif action == 'add':
+        # filter out repos with this Collection in their denylist
+        # TODO ^
+
+        # dispatch copy tasks
+        for repo in org_repos:
+            locks = [repo]
+            task_args = (collection_version_pk, repo.pk)
+            enqueue_with_reservation(add_content_to_repository, locks, args=task_args)
```

Comment:
What all ends up in the db transactions?  Feels like all the changes to a org repo should be in a single transaction.